nerfstudio/cameras/cameras.py

```diff
@@ -588,8 +590,15 @@ def _generate_rays_from_coords(
         elif distortion_params_delta is not None:
             distortion_params = distortion_params_delta
 
+        # Do not apply distortion for equirectangular images
         if distortion_params is not None:
-            coord_stack = camera_utils.radial_and_tangential_undistort(coord_stack, distortion_params)
+            mask = (self.camera_type[true_indices] != CameraType.EQUIRECTANGULAR.value).squeeze(-1)  # (num_rays)
+            coord_mask = torch.stack([mask, mask, mask], dim=0)
+            if mask.any():
+                coord_stack[coord_mask, :] = camera_utils.radial_and_tangential_undistort(
+                    coord_stack[coord_mask, :].reshape(3, -1, 2),
+                    distortion_params[mask, :],
+                ).reshape(-1, 2)
```

Comment:
@ethanweber I saw your refactor of `Cameras`. Before I merge this PR, can you take a quick look at this piece to make sure I'm not royally messing anything up?

Reply:
I think it looks correct to me. The logic checks out as far as I can tell.